rust/src/array.rs

```diff
@@ -285,6 +282,123 @@ def_primitive_array!(DataType::Int64, i64);
 def_primitive_array!(DataType::Float32, f32);
 def_primitive_array!(DataType::Float64, f64);
 
+/// Array whose elements are of boolean types.
+pub struct BooleanArray {
+    data: ArrayDataRef,
+    raw_values: RawPtrBox<u8>,
+}
+
+impl BooleanArray {
+    pub fn new(length: i64, values: Buffer, null_count: i64, offset: i64) -> Self {
+        let array_data = ArrayData::builder(DataType::Boolean)
+            .len(length)
+            .add_buffer(values)
+            .null_count(null_count)
+            .offset(offset)
+            .build();
+        BooleanArray::from(array_data)
+    }
+
+    /// Returns a `Buffer` holds all the values of this array.
+    ///
+    /// Note this doesn't take account into the offset of this array.
```

Comment:
Shouldn't this be something like "Note this does not take the offset of the underlying `ArrayData` into account"

Reply:
Yes offset technically is a property of `ArrayData`. However it is also (indirectly) a property of a `Array` since `ArrayData` encapsulates the common properties for all arrays. So, IMO this comment should be fine.

Reply:
ok, sounds good